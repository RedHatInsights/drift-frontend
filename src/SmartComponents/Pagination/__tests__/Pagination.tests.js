import React from 'react';
import { shallow, mount } from 'enzyme';

import TablePagination from '../Pagination';
import toJson from 'enzyme-to-json';

describe('table pagination', () => {
    let props;

    beforeEach(() => {
        props = {
            page: 1,
            perPage: 50,
            total: 0,
            isCompact: false
        };
    });

    it('should render correctly', async () => {
        const wrapper = shallow(
            <TablePagination { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render pagination compact', () => {
        props.isCompact = true;

        const wrapper = shallow(
            <TablePagination { ...props } />
        );

        expect(wrapper.prop('isCompact')).toBe(true);
    });

    it('should call updatePagination with page 2 and perPage 50', () => {
        props.total = 100;
        const updatePagination = jest.fn();
        const wrapper = mount(
            <TablePagination
                { ...props }
                updatePagination={ updatePagination }
            />
        );

        wrapper.find('.pf-c-button').at(2).simulate('click');
        expect(updatePagination).toHaveBeenCalledWith({ page: 2, perPage: 50 });
    });

    it('should call updatePagination on baselines table with page 2 and perPage 50', () => {
        props.total = 100;
        const updatePagination = jest.fn();
        const wrapper = mount(
            <TablePagination
                { ...props }
                updatePagination={ updatePagination }
                tableId={ 'COMPARISON' }
            />
        );

        wrapper.find('.pf-c-button').at(2).simulate('click');
        expect(updatePagination).toHaveBeenCalledWith({ page: 2, perPage: 50 }, 'COMPARISON');
    });

    it('should call updatePagination with page: 1 perPage 10', () => {
        props.total = 100;
        const updatePagination = jest.fn();
        const wrapper = mount(
            <TablePagination
                { ...props }
                updatePagination={ updatePagination }
            />
        );

        wrapper.find('.pf-c-options-menu__toggle-button').at(0).simulate('click');
        wrapper.find('.pf-c-options-menu__menu-item').at(0).simulate('click');
        expect(updatePagination).toHaveBeenCalledWith({ page: 1, perPage: 10 });
    });

    it('should call updatePagination on baselines table with page: 1 perPage 10', () => {
        props.total = 100;
        const updatePagination = jest.fn();
        const wrapper = mount(
            <TablePagination
                { ...props }
                updatePagination={ updatePagination }
                tableId={ 'COMPARISON' }
            />
        );

        wrapper.find('.pf-c-options-menu__toggle-button').at(0).simulate('click');
        wrapper.find('.pf-c-options-menu__menu-item').at(0).simulate('click');
        expect(updatePagination).toHaveBeenCalledWith({ page: 1, perPage: 10 }, 'COMPARISON');
    });
});
