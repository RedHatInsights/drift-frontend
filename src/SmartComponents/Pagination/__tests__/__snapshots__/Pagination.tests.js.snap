// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`table pagination should render correctly 1`] = `
<Pagination
  className=""
  defaultToFullPage={false}
  firstPage={1}
  isCompact={false}
  isDisabled={false}
  isSticky={false}
  itemCount={0}
  itemsEnd={null}
  itemsStart={null}
  offset={0}
  onFirstClick={[Function]}
  onLastClick={[Function]}
  onNextClick={[Function]}
  onPageInput={[Function]}
  onPerPageSelect={[Function]}
  onPreviousClick={[Function]}
  onSetPage={[Function]}
  ouiaSafe={true}
  page={0}
  perPage={50}
  perPageOptions={
    Array [
      Object {
        "title": "10",
        "value": 10,
      },
      Object {
        "title": "20",
        "value": 20,
      },
      Object {
        "title": "50",
        "value": 50,
      },
      Object {
        "title": "100",
        "value": 100,
      },
    ]
  }
  titles={
    Object {
      "currPage": "Current page",
      "items": "",
      "itemsPerPage": "Items per page",
      "ofWord": "of",
      "optionsToggle": "Items per page",
      "page": "",
      "paginationTitle": "Pagination",
      "perPageSuffix": "per page",
      "toFirstPage": "Go to first page",
      "toLastPage": "Go to last page",
      "toNextPage": "Go to next page",
      "toPreviousPage": "Go to previous page",
    }
  }
  toggleTemplate={[Function]}
  variant="top"
  widgetId="pagination-options-menu"
/>
`;
